apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -174,8 +174,10 @@ impl Plugin for Telemetry {
         let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
         Self::replace_tracer_provider(tracer_provider);
 
-        replace_layer(Box::new(telemetry))
-            .expect("set_global_subscriber() was not called at startup, fatal");
+        let result = replace_layer(Box::new(telemetry));
+        if cfg!(not(test)) {
+            result.expect("set_global_subscriber() was not called at startup, fatal");
+        }
```

Comment:
This might be overlapping with @Geal s work?

Reply:
this should not be neccessary once #1463 lands. And with the current version of the code it smells like a workaround for a deeper issue :upside_down_face: 
so idk, either dig into it or wait for the fix :grin: